### ISSUE TITLE
How do I create an alias for an generic type bound to some typevars?

### ISSUE THREAD

**Issue by NeilGirdhar** [python/mypy#9574]
I'd like to use a shorthand for a bound generic type:
```python
from typing import TypeVar, Generic

T = TypeVar('T')

class X(Generic[T]):
    pass

XT = X[T]

class Y(XT, Generic[T]):
    def f(self, xt: XT) -> XT:
        pass
```
Unfortunately `mypy --disallow-incomplete-defs` complains:
```
a.py:10: error: Missing type parameters for generic type "XT"
a.py:11: error: Missing type parameters for generic type "XT"
```

**Comment by hauntsaninja** [comment]
I don't think mypy is going to support this. `T` is unbound when you define `XT`, so there's no way to tell it apart from a generic type alias (without running afoul of PEP 484 and the reasons for why PEP 484 is the way it is). I think any implementation of this would in practice mostly just result in false negatives for typevar misuse.
As with any feature, we can reconsider if there are more numerous or compelling requests for it, but as it stands I'm closing.

**Comment by NeilGirdhar** [comment]
I understand your point, but as generic types have increasing numbers of parameters, there has to be some way to find a few of them and return a partially bound type.  If the syntax I showed is impossible, can't there be some way of doing a partial (typing) function application?  Maybe:
```
from typing import partial_type, TypeVar

T = TypeVar('T')
XT = partial_type(X, T)  # Bind first type parameter to T
YT = partial_type(Y, T=T)  # Bind named type parameter T to T
# etc.
```

If we can't bind shorthands, then when these types get really involved, we're force to copy and paste large type specifications.  In my case, I have [five generic type parameters](https://github.com/NeilGirdhar/tjax/blob/master/tjax/fixed_point/iterated_function.py) in some cases.  It makes my library really hard to read when I want to create shorthands.
